core/src/main/java/org/elasticsearch/index/shard/ReplicationGroup.java

```diff
@@ -43,6 +77,28 @@ public IndexShardRoutingTable getRoutingTable() {
         return inSyncAllocationIds;
     }
 
+    /**
+     * Returns the set of shard allocation ids that are in the in-sync set but have no assigned routing entry
+     */
+    public Set<String> getUnavailableInSyncShards() {
+        return unavailableInSyncShards;
+    }
+
+    /**
+     * Returns the subset of shards in the routing table that should be replicated to. Includes relocation targets.
+     */
+    public List<ShardRouting> getReplicationTargets() {
+        return replicationTargets;
+    }
+
+    /**
+     * Returns the subset of shards in the routing table that are unassigned or not required to replicate to. Includes relocation targets.
```

Comment:
can you please add a concrete description of what these shards? i.e., unassigned shards or initializing shards that are who are still being built and are not ready to receive operations.